lib/logger.js

```diff
@@ -6,449 +8,449 @@
-            message.app = opts.app || message.app;
-            message.env = opts.env || message.env;
-
-            if (opts.timestamp && isValidTimestamp(opts.timestamp)) {
-                message.timestamp = opts.timestamp;
-            }
-
-            if (opts.context && !opts.meta) {
-                opts.meta = opts.context;
-            }
-
-            if (typeof opts.meta === 'object') {
-                opts.meta = Object.assign({}, message.meta, opts.meta);
-                if (opts.index_meta || (opts.index_meta === undefined && this._index_meta)) {
-                    message.meta = opts.meta;
-                } else {
-                    message.meta = stringify(opts.meta);
-                }
-            }
-
-            if (opts.logSourceCRN) {
-                message.logSourceCRN = opts.logSourceCRN;
-            }
-
-            if (opts.saveServiceCopy) {
-                message.saveServiceCopy = opts.saveServiceCopy;
-            }
-
-            if (opts.appOverride) {
-                message.appOverride = opts.appOverride;
-            }
-
-            if (this._shimProperties) {
-                this._shimProperties.forEach((prop) => {
-                    if (opts.hasOwnProperty(prop)) {
-                        message[prop] = opts[prop];
-                    }
-                });
-            }
+          message.meta = stringify(opts.meta)
         }
-    } else if (!this._index_meta) {
-        message.meta = stringify(message.meta);
-    }
-
-    if (this._err) {
-        return this._err;
-    }
+      }
 
-    this._bufferLog(message, callback);
-};
-
-Logger.prototype._bufferLog = function(message, callback) {
-    if (!message || !message.line) {
-        debug('Ignoring empty message');
-        return;
-    }
-
-    if (!callback || typeof callback !== 'function') {
-        callback = (err) => { debug(err); };
-    }
+      if (opts.logSourceCRN) {
+        message.logSourceCRN = opts.logSourceCRN
+      }
 
-    if (this._max_length && message.line.length > configs.MAX_LINE_LENGTH) {
-        message.line = message.line.substring(0, configs.MAX_LINE_LENGTH) + ' (cut off, too long...)';
-        debug('Line was longer than ' + configs.MAX_LINE_LENGTH + ' chars and was truncated.');
-    }
+      if (opts.saveServiceCopy) {
+        message.saveServiceCopy = opts.saveServiceCopy
+      }
 
-    this._bufByteLength += sizeof(message);
+      if (opts.appOverride) {
+        message.appOverride = opts.appOverride
+      }
 
-    if (this._bufByteLength >= this._failedBufRetentionLimit) {
-        debug(`buffer size exceeded the failed buffer retention limit. Dropping this line: ${message.line}`);
-        this._bufByteLength -= sizeof(message);
-    } else {
-        debug('Buffering message: %s', message.line);
-        this._buf.push(message);
+      if (this._shimProperties) {
+        this._shimProperties.forEach((prop) => {
+          if (opts.hasOwnProperty(prop)) {
```

Comment:
last time, we saw somewhere that using `.hasOwnProperty` like this is risky and there should be safer way - [this](https://www.nickang.com/2018-02-04-developers-use-object-prototype-hasownproperty-call/) explains why it's risky

Reply:
Yes.  I was saving that for the major refactor.  I tried to change as little as possible with this in order to keep it a patch.  It's in here: https://logdna.atlassian.net/browse/LOG-6626

Reply:
Sounds good!